contracts/libraries/LibSortValidatorsByBeacon.sol

```diff
@@ -0,0 +1,235 @@
+      // filter and pick governance validators
+      (uint256[] memory gvIds,) = indices.filterBy(trustedWeights, governanceValidatorFilter);
+      // pick top `nGV` governance validators
+      (gvIds,) = gvIds.pickTopKByValues(stakeds.take(gvIds), nGV);
+      // mark the existence of governance validators
+      bitmap.setBatch(gvIds);
+
+      // filter and pick standard validators
+      // The number of governance validators is fewer than the maximum governance validator,
+      // The remainning number is transfered to standard validator case
+      nSV += nGV - gvIds.length;
+      (uint256[] memory svIds,) = indices.filterBy(bitmap.inner, standardValidatorFilter);
+      // pick top `nSV` standard validators
+      (svIds,) = svIds.pickTopKByValues(stakeds.take(svIds), nSV);
+      // mark the existence of standard validators
+      bitmap.setBatch(svIds);
+
+      // filter and pick non-standard validators
+      (uint256[] memory rvIds,) = indices.filterBy(bitmap.inner, nonStandardValidatorFilter);
+
+      // save the sorted validator set
+      _saveSorted({
+        nRV: nRV,
+        period: period,
+        stakeds: stakeds.take(rvIds),
+        rotatingValidators: cids.take(rvIds),
+        nonRotatingValidators: cids.take(gvIds.concat(svIds))
+      });
+    }
+  }
+
+  /**
+   * @dev Saves the unsorted IDs for a given period.
+   * @param period The period for which the IDs are being saved.
+   * @param ids An array of addresses representing the unsorted IDs.
+   */
+  function _saveUnsorted(uint256 period, address[] memory ids) private {
+    Validator storage $ = _getValidatorPerPeriodLocation()[period];
+    $.savedAll = true;
+    $.unsorted.ids = ids;
+  }
+
+  /**
+   * @dev Saves the sorted validators for a given period, beacon, and number of non-rotating validators.
+   * @param period The period for which the validators are being saved.
+   * @param nRV The number of non-standard validators.
+   * @param nonRotatingValidators An array of addresses representing the non-rotating validators.
+   * @param rotatingValidators An array of addresses representing the rotating validators.
+   * @param stakeds An array of stake values corresponding to the rotating validators.
+   */
+  function _saveSorted(
+    uint256 period,
+    uint256 nRV,
+    address[] memory nonRotatingValidators,
+    address[] memory rotatingValidators,
+    uint256[] memory stakeds
+  ) private {
+    SortedValidator storage $ = _getValidatorPerPeriodLocation()[period].sorted;
+
+    // save the non-rotating validators
+    $.nonRotatingValidators = nonRotatingValidators;
+
+    if (nRV == 0) return;
+
+    $.nRV = uint16(nRV);
+    delete $.rotatingValidators;
+
+    // pack `rotatingValidators` and `stakeds` into `RotatingValidator` struct (which cost 1 slot) to save gas
+    // max cap of RON is 1 billion, so 96 bits (~80 billion) are enough
+    uint96[] memory downCastedStakeds = stakeds.unsafeToUint96s();
```

Comment:
```suggestion
    uint96[] memory narrowingCastingStakeds = stakeds.unsafeToUint96s();
```

https://learn.microsoft.com/en-us/dotnet/visual-basic/programming-guide/language-features/data-types/widening-and-narrowing-conversions